### PR TITLE
refactor: 💡 Updated Falso generator to support JSON repo / FakeOption

### PR DESCRIPTION
**Pull request by DaanDeSmedt**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngneat/falso/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

The falso workspace generator did not :

- create a matching `.json` file for the newly to add falso
- declare function with `FakeOptions` parameter / type

## What is the new behavior?

The falso workspace generate will :

- generating a matchin falso JSON file according `--json` option value (default : `false`) 
- support `FakeOptions` to function definition

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: tools/generators/falso/index.ts [hunk 1]
```diff
@@ -2,21 +2,40 @@ import { Tree, getProjects, formatFiles, names } from '@nrwl/devkit';
...
   if (sourceRoot) {
-    tree.write(join(sourceRoot, 'lib', `${n.fileName}.ts`), ` 
-      import { rand } from './core';
-      
-      export function ${n.propertyName}() {
-        return rand([
-        ])
+    if (json) {
+      tree.write(
+        join(sourceRoot, 'lib', `${n.fileName}.json`),
```
Comment (NetanelBasal) [inline tools/generators/falso/index.ts:16]: Create a function for the base path and use in in both places, please

File: tools/generators/falso/index.ts [hunk 2]
```diff
@@ -2,21 +2,40 @@ import { Tree, getProjects, formatFiles, names } from '@nrwl/devkit';
...
+      join(sourceRoot, 'lib', `${n.fileName}.ts`),
+      `
+      import { FakeOptions, fake } from './core';
+      ${json ? `import { data } from './${n.fileName}.json'` : ''}
+
+      export function ${
+        n.propertyName
+      }<Options extends FakeOptions>(options?: Options) {
+        return fake(${json ? 'data' : '[]'}, options);
       }
```
Comment (NetanelBasal) [inline tools/generators/falso/index.ts:36]: Please add also a spec file.

Comment (DaanDeSmedt) [reply tools/generators/falso/index.ts:36]: Meaning to also generate the `.spec` file for the to generate falso right?
e.g. `yarn new --name dummy --json` produces `.../tests/dummy.spec.ts` right?

Do we want a explicit option to include the spec file (`--test`) ?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -3,8 +3,11 @@
 🙏 We would ❤️ for you to contribute to Falso and help make it even better than it is today!
 
 ## Developing
+
 - Run `npm i`
-- Create new falso by running `yarn f name`
+- Create new falso by running `yarn new --name name --json`
+
+  - Use `--json` (default `false`) to indicate that the falso will load it's data from a JSON file
```
Comment (Technologeek) [inline CONTRIBUTING.md:10]: Just a question here: Won't all falso's in the future will have to serve data from a static JSON file?

Comment (NetanelBasal) [reply CONTRIBUTING.md:10]: No, many of the functions will generate the data programmatically

File: tools/generators/falso/index.ts (hunk 2 above)
Comment (DaanDeSmedt) [reply tools/generators/falso/index.ts:34]: Falso generator will create a matching spec file by default.
Added `--skip-test` to change default behaviour.

**Pull request merged**